### PR TITLE
Milestone4: Add Join-tables.

### PR DESCRIPTION
**Pull request by carlylechia**
To complete this milestone, I

- [x] Updated the schema as requested, and made changes to the `schema.sql file`
- [x] Inserted the required data into their corresponding tables as seen in the `data.sql file`
- [x] Queried all the necessary tables and got answers to all questions asked in the [project requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_join_table.md), and made changes to the `queries.sql file`.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -13,3 +13,7 @@ ALTER TABLE animals ADD species_id integer;
 ALTER TABLE animals ADD CONSTRAINT FK_animalspecies FOREIGN KEY(species_id) REFERENCES species(id);
 ALTER TABLE animals ADD owner_id integer;
 ALTER TABLE animals ADD CONSTRAINT FK_animalowners FOREIGN KEY(owner_id) REFERENCES owners(id);
+
+CREATE TABLE vets(id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY, name VARCHAR(100), age INT, date_of_graduation date);
```
Comment (mateo951) [inline schema.sql:17]: Well done ✔ `vets` table is created properly 🔥

File: data.sql [hunk 2]
```diff
@@ -37,3 +37,7 @@ UPDATE animals SET owner_id = 2 WHERE name IN ('Gabumon', 'Pikachu');
 UPDATE animals SET owner_id = 3 WHERE name IN ('Plantmon', 'Devimon');
 UPDATE animals SET owner_id = 4 WHERE name IN ('Charmander', 'Squirtle','Blossom');
 UPDATE animals SET owner_id = 5 WHERE name IN ('Angemon', 'Boarmon');
+
+INSERT INTO vets(name, age, date_of_graduation) VALUES('William Tatcher', 45, '2000-04-23'),('Maisy Smith', 26, '2019-01-17'),('Stephanie Mendez', 64, '1981-05-04'),('Jack Harkness', 38, '2008-06-08');
+INSERT INTO specializations VALUES(1,1),(3,2),(3,1),(4,2);
+INSERT INTO visits VALUES(1,1,'2020-05-24'),(1,3,'2020-07-22'),(2,4,'2021-02-02'),(3,2,'2020-01-05'),(3,2,'2020-03-08'),(3,2,'2020-05-14'),(4,3,'2021-05-04'),(5,4,'2021-02-24'),(6,2,'2019-12-21'),(6,1,'2020-08-10'),(6,2,'2021-04-07'),(7,3,'2019-09-29'),(8,4,'2020-10-03'),(8,4,'2020-11-04'),(9,2,'2019-01-24'),(9,2,'2019-05-15'),(9,2,'2020-02-27'),(9,2,'2020-08-03'),(10,3,'2020-05-24'),(10,1,'2021-01-11');
```
Comment (mateo951) [inline data.sql:43]: Nice! All values inserted look good ✔

File: queries.sql [hunk 3]
```diff
@@ -78,3 +78,81 @@ SELECT owners.full_name, COUNT(animals.name) FROM owners JOIN animals ON owners.
...
+--          3 |      2 | 2020-03-08
+--          9 |      2 | 2020-02-27
+--          3 |      2 | 2020-01-05
+--          6 |      2 | 2019-12-21
+--          7 |      3 | 2019-09-29
+--          9 |      2 | 2019-05-15
+--          9 |      2 | 2019-01-24
+SELECT vets.name, COUNT(vets.name) AS num_of_visits FROM vets LEFT JOIN specializations ON vets.id = specializations.vet_id JOIN visits ON vets.id = visits.vet_id WHERE specializations.species_id IS NULL GROUP BY vets.name ORDER BY COUNT(vets.name) DESC;
+
+SELECT species.name AS expected_specialty FROM animals JOIN visits ON animals.id = visits.animal_id JOIN vets ON vets.id = visits.vet_id JOIN species ON species.id = animals.species_id WHERE vets.name = 'Maisy Smith' GROUP BY species.name ORDER BY COUNT(DISTINCT animals.name) DESC LIMIT 1;
```
Comment (mateo951) [inline queries.sql:155]: Great! All of your queries respond the questions 📝

**Review by mateo951 — APPROVED** [review]
### Hi @carlylechia, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 📝 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 3 above)
Comment (carlylechia) [reply queries.sql:155]: Thank you @mateo951

**Pull request merged**